Refactoring positive integers (#1059)
UniMath/agda-unimath: src/elementary-number-theory/strict-inequality-integer-fractions.lagda.md

open import elementary-number-theory.inequality-integers
open import elementary-number-theory.integer-fractions
open import elementary-number-theory.integers
open import elementary-number-theory.mediant-integer-fractions
open import elementary-number-theory.multiplication-integers
open import elementary-number-theory.multiplication-positive-and-negative-integers
open import elementary-number-theory.nonnegative-integers
open import elementary-number-theory.nonpositive-integers
open import elementary-number-theory.positive-and-negative-integers
open import elementary-number-theory.positive-integers
open import elementary-number-theory.strict-inequality-integers

open import foundation.action-on-identifications-functions
open import foundation.cartesian-product-types
open import foundation.coproduct-types
open import foundation.decidable-propositions
open import foundation.dependent-pair-types
open import foundation.function-types
open import foundation.identity-types
open import foundation.negation
open import foundation.propositions
open import foundation.transport-along-identifications
open import foundation.universe-levels
```

</details>

## Idea

> A fraction `m/n` is strictly less than a fraction `m'/n'` iff `m * n'` is
> strictly less than `m' * n`.

Reviewer: ```suggestion
An [integer fraction](elementary-number-theory.integer-fractions.md) `m/n` is _strictly less than_ the fraction `m'/n'` if the [integer product](elementary-number-theory.multiplication-integers.md) `m * n'` is
[strictly less](elementary-number-theory.strict-inequality-integers.md) than `m' * n`.
```
It's conventional to write "if" when defining a term, even though it does form a logical equivalence.

          ( H')))
      ( is-positive-denominator-fraction-ℤ q)

  concatenate-le-sim-fraction-ℤ :
    le-fraction-ℤ p q →
    sim-fraction-ℤ q r →
    le-fraction-ℤ p r
  concatenate-le-sim-fraction-ℤ H H' =
    is-positive-right-factor-mul-ℤ
      ( is-positive-eq-ℤ
        ( inv ( lemma-right-sim-cross-mul-diff-fraction-ℤ p q r H'))
        ( is-positive-mul-ℤ
          ( is-positive-denominator-fraction-ℤ r)
          ( H)))
      ( is-positive-denominator-fraction-ℤ q)
```

### Fractions with equal denominator compare the same as their numerators

```agda
module _
  (x y : fraction-ℤ) (H : denominator-fraction-ℤ x ＝ denominator-fraction-ℤ y)
  where

  le-fraction-le-numerator-fraction-ℤ :
    le-ℤ (numerator-fraction-ℤ x) (numerator-fraction-ℤ y) →
    le-fraction-ℤ x y
  le-fraction-le-numerator-fraction-ℤ H' =
    tr
      ( λ (k : ℤ) →
        le-ℤ
          ( numerator-fraction-ℤ x *ℤ k)
          ( numerator-fraction-ℤ y *ℤ denominator-fraction-ℤ x))
      ( H)
      ( preserves-le-left-mul-positive-ℤ
        ( positive-denominator-fraction-ℤ x)
        ( numerator-fraction-ℤ x)
        ( numerator-fraction-ℤ y)
        ( H'))
```

> ### The mediant of two fractions is between them

Reviewer: ```suggestion
### The mediant of two distinct fractions lies strictly between them
```

> # Strict inequality on integer fractions

Reviewer: ```suggestion
# Strict inequality on the integer fractions
```
